File: 프로그래머스/2/42584. 주식가격/주식가격.py
def solution(prices):
    N = len(prices)
    answer = [0] * N
    for i in range(N):
        count = 0
        for j in range(i + 1, N):
            count += 1
            if prices[i] > prices[j]:
                break
        answer[i] = count
    return answer 

File: 프로그래머스/2/42584. 주식가격/주식가격.java
import java.util.*;

class Solution {
    public int[] solution(int[] prices) {
        int N = prices.length;
        int[] answer = new int[N];
        Stack<Integer> idxStack = new Stack<>();
        
        for (int i = 0; i < N; i++) {
            while (!idxStack.isEmpty() && prices[idxStack.peek()] > prices[i]) {
                int top = idxStack.pop();
                answer[top] = i - top;
            }
            
            idxStack.push(i);
        }
        
        while (!idxStack.isEmpty()) {
            int top = idxStack.pop();
            answer[top] = (N - 1) - top;
        }
        
        return answer;
    }
}